include/cantera/cython/funcWrapper.h

```diff
@@ -96,6 +96,100 @@ extern "C" {
     CANTERA_CAPI PyObject* pyCanteraError;
 }
 
+
+//! A class to hold information needed to call Python functions from delgated
```

Comment:
It appears that `PyFuncInfo` is an augmented version of `Func1Py` ... I guess many of my previous comments about this PR were pointing in the direction that having a more powerful/flexible `Func1` would be very desirable even outside of delegated methods.

Reply:
Yes, this is essentially a generalization of `Func1Py`, using `std::function` to wrap the Python function rather than `Func1` (which only supports a single function signature). This certainly could be used outside the context of the `Delegator` class, but we don't currently have a lot of capabilities that take functions more complex than `Func1` as input.

Reply:
As implied in #1101 this could be useful for a more generic `CustomReaction`. For this relatively simple class, simply providing a function is sufficient. Of course, this could be coupled with a `Delegator` ... there likely is an efficient solution. The boilerplate of the current implementation is simply `CustomRate.set_rate_function`, which is really simple to use.